### PR TITLE
LibRegex: Implement section B.1.4. of the ECMA262 spec / LibRegex: some tiny fixes

### LINKED ISSUE #5517: LibRegex: Parsing error in prettify.js
From: https://getbootstrap.com/2.0.0/assets/js/google-code-prettify/prettify.js
Error:
```
WebContent(30): HTMLScriptElement: Running script assets/js/google-code-prettify/prettify.js
WebContent(30): Throwing JavaScript Error: SyntaxError, RegExp compile error: Error during parsing of regular expression:
    ^(?:^^\.?|[!+-]|!=|!==|#|%|%=|&|&&|&&=|&=|\(|\*|\*=|\+=|,|-=|->|\/|\/=|:|::|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|[?@[^]|\^=|\^\^|\^\^=|{|\||\|=|\|\||\|\|=|~|break|case|continue|delete|do|else|finally|instanceof|return|throw|try|typeof)\s*(\/(?=[^*/])(?:[^/[\\]|\\[\S\s]|\[(?:[^\\\]]|\\[\S\s])*(?:]|$))+\/)
                                                                                                                                                  ^---- Invalid regular expression.
```

### LINKED ISSUE #5518: LibRegex: Parsing error with unlimited quantifiers
For example, `([0-9a-f]{3,})` should match 3 _or more_ times.
The parser doesn't handle this:
```
WebContent(30): Throwing JavaScript Error: SyntaxError, RegExp compile error: Error during parsing of regular expression:
    (?:#|\^)([0-9a-f]{3,})\:(?:#|\^)([0-9a-f]{3,})
                        ^---- Content of {} invalid: not a number, number too large, more than two numbers, first larger than second.
```
(From https://getbootstrap.com/docs/3.3/assets/js/docs.min.js)

### PR DESCRIPTION
**Pull request by alimpfard**
This allows the parser to deal with crazy patterns like the one in #5517.

I decided to _conditionally_ bite the bullet (in the regex library).
As discussed, this enables that flag by default in LibJS (+a FIXME). we can later disable it as needed.

Fixes #5517.
Fixes #5518.

### CODE REVIEW

**Comment by alimpfard** [comment]
cc @linusg for JS stuff

File: Userland/Libraries/LibRegex/RegexParser.h [hunk 1]
```diff
@@ -197,6 +201,17 @@ class ECMA262Parser final : public Parser {
...
+    // Used only by B.1.4, Regular Expression Patterns (Extended for use in browsers)
+    bool parse_quantifiable_assertion(ByteCode&, size_t&, bool named);
+    bool parse_extended_atom(ByteCode&, size_t&, bool named);
+    bool parse_inner_disjunction(ByteCode& bytecode_stack, size_t& length, bool unicode, bool named);
+    bool parse_invalid_braced_quantifier(); // Note: This function either parses and *fails*, or doesn't parse anything and returns false.
+    bool parse_legacy_octal_escape_sequence(ByteCode& bytecode_stack, size_t& length);
+    Optional<u8> parse_legacy_octal_escape();
+
+    // Let's keep them separate, for now?
+    bool m_should_use_browser_extended_grammar { false };
```
Comment (linusg) [inline Userland/Libraries/LibRegex/RegexParser.h:214]: Yes, definitely. What I'd like to do, and I might have mentioned this on IRC before, is adding an Annex B flag to the JS parser and engine, so you can conditionally enable/disable all the legacy stuff - I'd say for now just always enable this and I'll come back to it, I'm sure there are other Annex B features already built in unconditionally.

The browser would have that always on, and the JS repl by default with an option to disable, or something like that.

**Review by linusg — COMMENTED** [review]
great stuff!

File: Userland/Libraries/LibRegex/RegexParser.h (hunk 1 above)
Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexParser.h:214]: Then I guess we can always pass the `Internal_BrowserExtended` flag from LibJS here, and any further changes can be done in LibJS.

File: Userland/Libraries/LibRegex/RegexByteCode.cpp [hunk 2]
```diff
@@ -701,21 +701,41 @@ const Vector<String> OpCode_Compare::variable_arguments_to_string(Optional<Match
...
+            if (!view.is_null() && view.length() > state().string_position) {
+                if (is_ascii) {
+                    result.empend(String::formatted(
+                        "compare against: '{}'",
+                        view.substring_view(compared_against_string_start_offset, state().string_position > view.length() ? 0 : 1).to_string()));
+                } else {
+                    auto str = view.substring_view(compared_against_string_start_offset, state().string_position > view.length() ? 0 : 1).to_string();
+                    u64 value = 0;
+                    if (str.length() >= sizeof(u64))
+                        value = *reinterpret_cast<const u64*>(str.characters());
```
Comment (tomuta) [inline Userland/Libraries/LibRegex/RegexByteCode.cpp:725]: Isn't this going to cause potentially unaligned access? Shouldn't this be done using `memcpy`?

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexByteCode.cpp:725]: Honestly, it's in debug code...
I was thinking this as I wrote it, then I thought "well it's debug code, so who cares if it blows up"...

I'll fix it when I feel less lazy, draft until then I guess

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexByteCode.cpp:725]: Hmmm, actually, why even bother, let's just dump out some hex instead.

Comment (tomuta) [reply Userland/Libraries/LibRegex/RegexByteCode.cpp]: I didn't even realize it was debug code. Anyway, on x86 this works with a slight speed penalty, though we should probably enable the alignment check at some point, which would then also lead to a crash. But it's not portable and not all architectures allow unaligned access. Older ARMs for example didn't allow it.

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexByteCode.cpp]: if we're debugging LibRegex on a non-x86 arch, I'll consider even the crash a win lol

Comment (tomuta) [reply Userland/Libraries/LibRegex/RegexByteCode.cpp]: Well wait until @awesomekling throws in the alignment check bit and it'll crash on x86 as well ;)

**Pull request merged**